### PR TITLE
TRAINTECH-328

### PR DESCRIPTION
**Pull request by josephoaks**
Powershell scripts to configure the Windows Server for the installation of the puppet-agent per request of partner trainers.

To use: 
 - Open Powershell
 - execute "[System.Net.ServicePointManager]::ServerCertificateValidationCallback={$true}"
 - execute "(New-Object System.Net.WebClient).DownloadFile("https://<ip address of master>:8140/packages/current/windows-x86_64/setup_windows.ps1", "C:\Users\Administrator\setup_windows.ps1")

 -execute "setup_windows.ps1"

The system will reboot, when the system returns
 - Open Powershell
 - execute setup_classroom.ps1

-- Files --
setup_windows.ps1 configures the server for install of puppet-agent
setup_classroom.ps1 installs and configures puppet-agent

### CODE REVIEW

**Comment by binford2k** [comment]
is this still being worked on? Is it superseded?

**Comment by josephoaks** [comment]
This has been superseded with the PR 129

File: manifests/master/windows.pp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  assert_private('This class should not be called directly')
+
+  File {
+    ensure => file,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0644',
+  }
+
+  $publicdir   = "/opt/puppetlabs/server/data/packages/public"
```
Comment (binford2k) [inline manifests/master/windows.pp]: why is this being calculated here instead of using https://github.com/puppetlabs/pltraining-classroom/blob/master/manifests/params.pp#L39

**Comment by josephoaks** [comment]
This was being obtained from params.pp#L39 however this did not set the correct path of /opt/puppetlabs/server/data/packages/public/2015.3.1/windows-x86_64 where the puppet-agent-x64.msi is located.

File: manifests/master/windows.pp (hunk 1 above)
Comment (binford2k) [inline manifests/master/windows.pp]: is there a reason why the script needs to live in the same directory as the MSI? for more context, the path you're linking here is owned by the pe_repo class, where the path that we use in the classroom is owned by us.

In any case, all of these config style variables need to come from params.pp

File: manifests/master/windows.pp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  $publicdir = $classroom::params::publicdir
+
+  File {
+    ensure => file,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0644',
+  }
+
+  $destination = "${publicdir}/${pe_server_version}/windows-x86_64"
```
Comment (samuelson) [inline manifests/master/windows.pp]: @josephoaks I think you need the "classroom" directory in here.

File: manifests/master/agent_tarball.pp [hunk 3]
```diff
@@ -9,7 +9,7 @@
   # https://pm.puppetlabs.com/puppet-agent/2015.2.0/1.2.2/repos/puppet-agent-el-6-x86_64.tar.gz
   $filename    = "puppet-agent-${::platform_tag}.tar.gz"
   $download    = "https://pm.puppetlabs.com/puppet-agent/${pe_server_version}/${aio_agent_version}/repos/${filename}"
-  $destination = "${publicdir}/${pe_server_version}/${aio_agent_version}/repos"
+  $destination = "${publicdir}/classroom/${pe_server_version}/${aio_agent_version}/repos"
```
Comment (samuelson) [inline manifests/master/agent_tarball.pp]: Why is this change here?

Comment (josephoaks) [inline manifests/master/agent_tarball.pp]: This was modified to use the ${publicdir} from the params, with the PS1 files it was trying to put them in the classroom directory that did not exist in fundamentals but does for architect.
